src/MyEventEmitter.js

```diff
@@ -1,12 +1,94 @@
 'use strict';
 
-export class MyEventEmitter {
-  on() {}
-  once() {}
-  off() {}
-  emit() {}
-  prependListener() {}
-  prependOnceListener() {}
-  removeAllListeners() {}
-  listenerCount() {}
+class MyEventEmitter {
+  constructor() {
+    this.events = {
+
+    };
+  }
+
+  addEvent(name, callback, repeat, prepend) {
+    if ({}.hasOwnProperty.call(this.events, name)) {
+      if (prepend) {
+        this.events[name].unshift({
+          repeat,
+          callback,
+        });
+      } else {
+        this.events[name].push({
+          repeat,
+          callback,
+        });
+      }
+    } else {
+      this.events[name] = [
+        {
+          repeat,
+          callback,
+        },
+      ];
+    }
+  }
+
+  on(name, callback) {
+    this.addEvent(name, callback, true, false);
+  }
+
+  once(name, callback) {
+    this.addEvent(name, callback, false, false);
+  }
+
+  prependListener(name, callback) {
+    this.addEvent(name, callback, true, true);
+  }
+
+  prependOnceListener(name, callback) {
+    this.addEvent(name, callback, false, true);
+  }
+
+  off(name, listener) {
+    if ({}.hasOwnProperty.call(this.events, name)) {
+      const callbackIndex = this.events[name].findIndex(
+        ({ callback }) => callback === listener
+      );
+
+      if (callbackIndex !== -1) {
+        if (this.events[name].length === 1) {
+          delete this.events[name];
+        } else {
+          this.events[name].splice(callbackIndex, 1);
+        }
+      }
+    }
+  }
+
+  emit(name, ...args) {
+    if ({}.hasOwnProperty.call(this.events, name)) {
+      const callbacks = this.events[name];
+
+      callbacks.forEach(({ callback }) => {
+        callback(args);
+      });
+      this.events[name] = callbacks.filter(({ repeat }) => repeat);
+    }
+  }
+
+  removeAllListeners() {
+    this.events = [];
```

Comment:
First of all, replace `this.events = [];` with `this.events = {};` to properly reset the events object. 

BUT, in task we should remove all listeners for a **specific event**, or all events if no event is specified. You did not check if it is a specific event, you just cleaned up all listeners for all events.